service/evidence/evidence_store.go

```diff
@@ -123,6 +125,11 @@
 
 		// Send response back to the client
 		err = stream.Send(res)
+
+		// Check for send errors
+		if errors.Is(err, io.EOF) {
```

Comment:
For me, only the sender can signal an EOF (and not the receiver since it just receives and can't now when EOF is reached). But maybe I oversee sth. in the streaming model.

Reply:
I was getting EOF gRPC errors in the benchmark tests.

Reply:
Ok, interesting. In the gRPC [stream example](https://github.com/grpc/grpc-go/blob/94ee3865e17c5645148f8d24a32dc50f40810f13/examples/route_guide/server/server.go#L125) they just return any error and maybe the EOF is hidden there.